proxy/grpcproxy/health.go

```diff
@@ -43,3 +51,25 @@ func checkHealth(c *clientv3.Client) etcdhttp.Health {
 	}
 	return h
 }
+
+func checkProxyHealth(c *clientv3.Client) etcdhttp.Health {
+	h := etcdhttp.Health{Health: "true"}
+	ctx, cancel := context.WithTimeout(c.Ctx(), time.Second)
+	_, err := c.Get(ctx, "a")
+	cancel()
+	if err != nil && err != rpctypes.ErrPermissionDenied {
+		h.Health = "false"
+		h.Reason = "GET ERROR"
```

Comment:
Should we include original error message?

Reply:
updated.thanks